compiler/src/dmd/hdrgen.d

```diff
@@ -134,37 +134,19 @@ void moduleToBuffer2(Module m, OutBuffer* buf, HdrGenState* hgs)
 
 private void statementToBuffer(Statement s, OutBuffer* buf, HdrGenState* hgs)
 {
-    scope v = new StatementPrettyPrintVisitor(buf, hgs);
-    s.accept(v);
-}
-
-private extern (C++) final class StatementPrettyPrintVisitor : Visitor
-{
-    alias visit = Visitor.visit;
-public:
-    OutBuffer* buf;
-    HdrGenState* hgs;
-
-    extern (D) this(OutBuffer* buf, HdrGenState* hgs) scope
+    void visitDefaultCase(Statement s)
     {
-        this.buf = buf;
-        this.hgs = hgs;
-    }
-
-    override void visit(Statement s)
-    {
-        buf.writestring("Statement::toCBuffer()");
-        buf.writenl();
+        s.error("visitDefaultCase() %d for %s", s.stmt, s.toChars());
```

Comment:
You shouldn't call `toChars` when there's a bug in `Statement.toChars`, it will result in endless recursion.

Reply:
Oops! Thanks for finding the source of the problem!